VueJS/index.js

```diff
@@ -1,13 +1,85 @@
-import List from './components/List.js'
-import ListItem from './components/ListItem.js'
 
-new Vue({
-  el: '#app',
-  data: {
-    todos: [
-      { id: '1', message: 'Foo', },
-      { id: '2', message: 'Bar', },
-      { id: '3', message: 'Baz', }
-    ]
-  }
+// import List from './components/List.js'
+// import ListItem from './components/ListItem.js'
+
+Vue.component('todo-list', {
+	props: ['todo', 'index'],
+	data: function() {
+		return {
+			setEdit: false,
+			temp: ''
+		}
+	},
+	template: `
+				<li>
+					<div v-if="!setEdit">
+						{{ todo.message }}
+						<button v-on:click="editing">Edit</button>
+						<button v-on:click="deleting">Delete</button>
+					</div>
+					<div v-else>
+						<input v-model="temp"/>
+						<button v-on:click="saving">Save</button>
+						<button v-on:click="canceling">Cancel</button>
+					</div>
+				</li>
+	`,
+	methods: {
+		editing: function() {
+			this.temp = this.todo.message;
+			this.setEdit = true;
+		},
+		deleting: function() {
+			this.setEdit = false;
+			vue.delete(this.index);
```

Comment:
Is it this: https://vuejs.org/v2/api/#Vue-delete ?

> you should rarely need to use it.

Does not sound like a best practice to me :thinking: